### ISSUE TITLE
TypeError when trying to create a lazy development server

### ISSUE THREAD

**Issue by langri-sha** [storybooks/storybook#4532]
### Bug or support request summary

When using the `@storybook/core` server middleware, configuring `{ devServer: { lazy: true } }` leads to a `TypeError`. 

### Steps to reproduce

- create a custom webpack configuration file in full control mode
- configure `storybookBaseConfig.devServer.lazy = true`
- create the middleware from `@storybook/core/dist/server/middleware`

### Please specify which version of Storybook and optionally any affected addons that you're running

- @storybook/react v4.0.0-rc.2

### Screenshots / Screencast / Code Snippets (Optional)

```
info => Loading presets
info => Loading custom .babelrc
info => Loading custom addons config.
info => Loading custom webpack config (full-control mode).
TypeError: Cannot read property 'toJson' of null
    at webpackDevMiddlewareInstance.waitUntilValid.stats (/home/filip/work/frontend3/node_modules/@storybook/core/dist/server/middleware.js:79:15)
    at ready (/home/filip/work/frontend3/node_modules/@storybook/core/node_modules/webpack-dev-middleware/lib/util.js:51:12)
    at Function.waitUntilValid (/home/filip/work/frontend3/node_modules/@storybook/core/node_modules/webpack-dev-middleware/index.js:97:7)
    at _default (/home/filip/work/frontend3/node_modules/@storybook/core/dist/server/middleware.js:69:32)
    at <anonymous>
```

**Comment by shilman** [comment]
@langri-sha Mind checking if this fixed in `^4.0.0` or `4.1.0-alpha.*`? There have been a bunch of changes since `rc.2`. Thanks for your patience and help!

**Comment by langri-sha** [comment]
Thanks for the heads up! Will try to take a look!
